clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -158,6 +161,39 @@ public void testCloseAll() {
         }
     }
 
+    @Test
+    public void testReadFully() throws IOException {
+        RandomAccessFile raf = new RandomAccessFile(TestUtils.tempFile(), "rw");
```

Comment:
If I use the code below to verify, the position does not change after channelMock.read. Where did I go wrong? Please kindly advice. Thanks.
`

        FileChannel channelMock = EasyMock.createMock(FileChannel.class);
        ByteBuffer byteBuffer = ByteBuffer.wrap("hello, world".getBytes());
        ByteBuffer buffer = ByteBuffer.allocate(12);
        EasyMock.expect(channelMock.write(byteBuffer)).andReturn(12);
        EasyMock.expect(channelMock.read(buffer, 0)).andReturn(12);
        EasyMock.replay(channelMock);
        channelMock.write(byteBuffer);
        channelMock.read(buffer, 0);
        System.out.println(buffer.position());  // always 0 :-(
        EasyMock.verify(channelMock);`